packages/sdk/src/document/document.ts

```diff
@@ -1180,11 +1180,26 @@ export class Document<T, P extends Indexable = Indexable> {
     }
 
     // NOTE(hackerwins): If the document has local changes, we need to apply
-    // them after applying the snapshot. We need to treat the local changes
-    // as remote changes because the application should apply the local
-    // changes to their own document.
+    // them after applying the snapshot, as local changes are not included in the snapshot data.
+    // Afterward, we should publish a snapshot event with the latest
+    // version of the document to ensure the user receives the most up-to-date snapshot.
     if (hasSnapshot) {
-      this.applyChanges(this.localChanges, OpSource.Remote);
+      this.applyChanges(this.localChanges, OpSource.Local);
+      this.publish([
+        {
+          type: DocEventType.Snapshot,
+          source: OpSource.Remote,
+          value: {
+            serverSeq: pack.getCheckpoint().getServerSeq().toString(),
+            snapshot: this.isEnableDevtools()
+              ? converter.bytesToHex(pack.getSnapshot()!)
+              : undefined,
+            snapshotVector: converter.versionVectorToHex(
+              pack.getVersionVector()!,
+            ),
+          },
+        },
+      ]);
```

Comment:
For consistency with other methods like `applyXXX` (`applyChanges`, `applyStatus`), could we modify `applySnapshot` to handle the localChanges reflection and event triggering internally?

Reply:
I completely agree with your opinion.

To maintain consistency, I think we can move the reflection of localChanges and event triggering internally. This way, we would need to pass `clientSeq` as a parameter.

However, the `applySnapshot` function is also called by `applyDocEvent`, as shown in the following code:
```typescript
if (event.type === DocEventType.Snapshot) {
  const { snapshot, serverSeq, snapshotVector } = event.value;
  if (!snapshot) return;
  this.applySnapshot(
    BigInt(serverSeq),
    converter.hexToVersionVector(snapshotVector),
    converter.hexToBytes(snapshot),
  );
  return;
}
```
I’m not sure, though, if `applyDocEvent` can pass enough parameters to reflect `localChanges` (perhaps `clientSeq`).

Could I get some advice?

Reply:
`applyDocEvent` is used to build documents based on DocEvents published by the `Client`. Since we're applying already generated events to the document, I think we shouldn't apply local changes. How about preventing local changes from being applied by using the maximum value for clientSeq?

Reply:
Currently, since the default value of `clientSeq` in `applySnapshot` is `-1`, all local changes remain without being removed and are applied after the snapshot. Initially, I thought local changes shouldn't be applied, but when building documents from DocEvents, there won't be any local changes, so this approach should work fine.

![image](https://github.com/user-attachments/assets/59a13fd7-0d09-4a1e-beef-be59b2915a31)